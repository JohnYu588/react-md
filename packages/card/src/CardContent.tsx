import type { HTMLAttributes } from "react";
import { forwardRef } from "react";
import type { CardContentClassNameOptions } from "./styles";
import { cardContent } from "./styles";

export interface CardContentProps
  extends HTMLAttributes<HTMLDivElement>,
    CardContentClassNameOptions {}

export const CardContent = forwardRef<HTMLDivElement, CardContentProps>(
  function CardContent(props, ref) {
    const {
      children,
      className,
      disablePadding,
      disableLastChildPadding,
      ...remaining
    } = props;

    return (
      <div
        {...remaining}
        ref={ref}
        className={cardContent({
          className,
          disablePadding,
          disableLastChildPadding,
        })}
      >
        {children}
      </div>
    );
  }
);
