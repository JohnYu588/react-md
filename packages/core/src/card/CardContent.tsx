import { forwardRef, type HTMLAttributes } from "react";
import { cardContent, type CardContentClassNameOptions } from "./styles.js";

/**
 * @since 6.0.0 Renamed `disableExtraPadding` to `disableLastChildPadding` and
 * removed the `disableParagraphMargin` prop.
 */
export interface CardContentProps
  extends HTMLAttributes<HTMLDivElement>,
    CardContentClassNameOptions {}

/**
 * **Server Component**
 *
 * @since 6.0.0 Renamed `disableExtraPadding` to `disableLastChildPadding` and
 * removed the `disableParagraphMargin` prop.
 */
export const CardContent = forwardRef<HTMLDivElement, CardContentProps>(
  function CardContent(props, ref) {
    const {
      children,
      className,
      disablePadding,
      disableSecondaryColor,
      disableLastChildPadding,
      ...remaining
    } = props;

    return (
      <div
        {...remaining}
        ref={ref}
        className={cardContent({
          className,
          disablePadding,
          disableSecondaryColor,
          disableLastChildPadding,
        })}
      >
        {children}
      </div>
    );
  }
);
