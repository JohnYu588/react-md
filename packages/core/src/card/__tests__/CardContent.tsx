import { describe, expect, it } from "@jest/globals";
import { createRef } from "react";
import { render, screen } from "../../test-utils/index.js";

import { CardContent } from "../CardContent.js";

describe("CardContent", () => {
  it("should apply the correct styling, HTML attributes, and allow a ref", () => {
    const ref = createRef<HTMLDivElement>();
    const props = {
      "data-testid": "content",
      ref,
      children: "Content",
    } as const;
    const { rerender } = render(<CardContent {...props} />);

    const card = screen.getByTestId("content");
    expect(ref.current).toBeInstanceOf(HTMLDivElement);
    expect(ref.current).toBe(card);
    expect(card).toMatchSnapshot();

    rerender(
      <CardContent
        {...props}
        style={{ color: "white" }}
        className="custom-class-name"
      />
    );
    expect(card).toMatchSnapshot();

    rerender(<CardContent {...props} disablePadding />);
    expect(card).toMatchSnapshot();

    rerender(<CardContent {...props} disableSecondaryColor />);
    expect(card).toMatchSnapshot();

    rerender(<CardContent {...props} disableLastChildPadding />);
    expect(card).toMatchSnapshot();
  });
});
