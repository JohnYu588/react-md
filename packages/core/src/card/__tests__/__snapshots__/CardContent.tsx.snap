// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent should apply the correct styling, HTML attributes, and allow a ref 1`] = `
<div
  class="rmd-card__content rmd-card__content--padded rmd-card__content--padding-bottom rmd-text-secondary-color"
  data-testid="content"
>
  Content
</div>
`;

exports[`CardContent should apply the correct styling, HTML attributes, and allow a ref 2`] = `
<div
  class="rmd-card__content rmd-card__content--padded rmd-card__content--padding-bottom rmd-text-secondary-color custom-class-name"
  data-testid="content"
  style="color: white;"
>
  Content
</div>
`;

exports[`CardContent should apply the correct styling, HTML attributes, and allow a ref 3`] = `
<div
  class="rmd-card__content rmd-card__content--padding-bottom rmd-text-secondary-color"
  data-testid="content"
  style=""
>
  Content
</div>
`;

exports[`CardContent should apply the correct styling, HTML attributes, and allow a ref 4`] = `
<div
  class="rmd-card__content rmd-card__content--padded rmd-card__content--padding-bottom"
  data-testid="content"
  style=""
>
  Content
</div>
`;

exports[`CardContent should apply the correct styling, HTML attributes, and allow a ref 5`] = `
<div
  class="rmd-card__content rmd-card__content--padded rmd-text-secondary-color"
  data-testid="content"
  style=""
>
  Content
</div>
`;
